### ISSUE TITLE
Check if ports are occupied before start sqlness test

### ISSUE THREAD

**Issue by waynexia** [GreptimeTeam/greptimedb#1602]
### What problem does the new feature solve?

Sqlness uses the common port (e.g. `4000`) to start server. It's common that the port is occupied by other process like another greptimedb process. In this case the test result will become weird because it's actually testing something else

### What does the feature do?

Check those ports before spawn subprocess. Sqlness only check if the port is up after spawn at present.

**Comment by Gump9** [comment]
I'm a little confused about 
> Check those ports before spawn subprocess. 
It that means Sqlness whill spawn subprocess?
Which file can give me a first look at the problem？
> Sqlness only check if the port is up after spawn at present.

**Comment by Gump9** [comment]
I'm a little confused about

>Check those ports before spawn subprocess.  

it that means Sqlness will spawn subprocess?  

Which file can give me a first look at the problem？
>Sqlness only check if the port is up after spawn at present.

**Comment by Gump9** [comment]
Sry,i don't see it has been assigned😭

**Comment by WenyXu** [comment]
> Sry,i don't see it has been assigned😭

Never mind. We got a ton of the good first issues waiting for you 🤪

**Comment by evenyag** [comment]
@Gump9 Are you still interested in this issue? Feel free to take it.

**Comment by Gump9** [comment]
> @Gump9 Are you still interested in this issue? Feel free to take it.

Yes,can I have a try?tks alot

**Comment by Gump9** [comment]
and i still want to know more details about this issue i [mentioned above](https://github.com/GreptimeTeam/greptimedb/issues/1602#issuecomment-1617255333)

**Comment by waynexia** [comment]
>it that means Sqlness will spawn subprocess?
>
>Which file can give me a first look at the problem？

Yes, sqlness runner will start greptimedb (metasrv/frontend/datanode) to test. You can check it out here https://github.com/GreptimeTeam/greptimedb/blob/fcff66e03904d80aacb91b8edd4e15240161d264/tests/runner/src/env.rs#L124

**Comment by Gump9** [comment]
@waynexia thx a lot😄

**Comment by Gump9** [comment]
@waynexia Hi!should i just panic like the code below？because i found "Sqlness only check if the port is up after spawn at present." do something like this:
```rust
if !util::check_port(check_ip_addr.parse().unwrap(), Duration::from_secs(10)).await {
            Env::stop_server(&mut process);
            panic!("{subcommand} doesn't up in 10 seconds, quit.")
}
```
or,should i choose another port and try to up a process again?

**Comment by waynexia** [comment]
`panic` should be enough. We can add one config entry for port later

**Comment by Gump9** [comment]
> `panic` should be enough. We can add one config entry for port later

thx!
